functions/api/users.js

```diff
@@ -166,4 +186,63 @@ function validateInput(body, type) {
     return validate(schema, body)
 }
 
-module.exports = router
\ No newline at end of file
+module.exports = router
+
+
+module.exports.modifyUsers = functions.firestore
+    .document(`/${constants.USERS}/{userId}`)
+    .onWrite(async (change, context) => {
+        if (!change.before._fieldsProto) {
+            await addUserToBranch(change.after)
+        } else if (!change.after._fieldsProto) {
+            await deleteUserFromBranch(change.before)
+        } else {
+            var oldData = change.before.data()
+            var newData = change.after.data()
+            if (oldData.branch !== newData.branch) {
+                await deleteUserFromBranch(change.before)
+                await addUserToBranch(change.after)
+            }
+        }
+    })
+
+async function addUserToBranch(user) {
+    try {
+        var { email, branch } = user.data()
+        const branchRef = db.doc(`${constants.BRANCHES}/${branch.toLocaleLowerCase()}`)
+        const doc = await branchRef.get()
+        // Check if branch is present in the collection
+        if (!doc.exists) {
+            throw new Error(`Branch ${branch} is not present in firestore!!!!`)
+        }
+        const users = doc.data().users
+        users.push(email)
```

Comment:
I am sorry, I misread your comment on this. The users array in the Branch should just contain firstname lastname not email.

Reply:
no problem. fixed it